src/frontend/get-github-configuration.ts

```diff
@@ -28,6 +30,16 @@ export default async (req: Request, res: Response, next: NextFunction): Promise<
         username: login,
         type: installation.target_type,
       });
+
+      const authedApp = await app.auth(installation.id);
+      enhanceOctokit(authedApp);
+
+      const repositories = await authedApp.paginate(
```

Comment:
are you getting all repos just to get the number?  You should maybe consider using the graphql endpoint so you don't have to do multiple calls.

Reply:
We should also consider using `Promise.all` so that all calls are parallelized instead of sequential. It'll reduce the amount of time to show a UI to the user.

Reply:
Not sure we can utilise graphql here. To get the number of repos we need to separate queries.

For orgs - the main account, so 'rachellerathbone' in my case, isn't 'considered an org so I get an error if I try to pass `installation.account.login` to this query. It does, however, give the right data back for other orgs:
```organization(login:$login) {
    repositories {
      totalCount
    }
  }```

For the primary account a second query is needed, however, it returns the incorrect data. It should return **1** (selected 1 repo) but it returns **35**. Not even sure where this number is coming from as I have 94 repos. Had a look at the list of repos coming back and they are completely random. Some I use frequently, others I haven't touched in years.
```user(login:$login) {
    repositories {
      totalCount
    }
  }```

Reply:
I think the proper way to do the query to the user is by using the `viewer` type which returns the authenticated user, in this case, the app at that specific installation ID:
```
query { 
  viewer { 
    repositories {
      totalCount
    }
  }
}
```

Try that to see if it returns the correct count. You can play around graphql with the explorer: https://docs.github.com/en/graphql/overview/explorer

Reply:
I just asked saquib about the discrepancies.  I've noticed a few myself between the REST api and graphql, which is extremely concerning :/

Reply:
Unfortunately the viewer query is just as whack and returns the wrong number for my primary account. Orgs are correct but 'rachellerathbone' returns the mysterious 35...

Reply:
that's really concerning... we really need to look more into this.

Reply:
Do we want to hold off on merging this in until we can use a graphql endpoint? Or make a note to update when graphql returns the correct data?

Reply:
yeah, probably not good to have this blocked because github can't get their APIs in order :P
Let's more forward without this and I'll look this up further as it will cause issues in the future.